lib/libc/Kconfig

```diff
@@ -28,6 +28,7 @@ config FULL_LIBC_SUPPORTED
 config MINIMAL_LIBC_SUPPORTED
 	bool
 	depends on !NATIVE_APPLICATION
+	depends on !REQUIRES_FULL_LIBC
```

Comment:
I don't think this is correct. MINIMAL_LIBC_SUPPORTED indicates the target itself supports the library.
REQUIRES_FULL_LIBC indicates an application or feature requires a full libc. Which has nothing to do with the target supporting or not minimallibc.

Reply:
should that board kconfig.defconfig have a
```
config MINIMAL_LIBC_SUPPORTED
	default n
```
?

Reply:
This would only solve the problem for this one board, but similar scenario may occur for others. I wanted to provide a generic solution and I thought that such dependency would be fine, as I saw that `PICOLIBC_SUPPORTED` has dependency on `!REQUIRES_FULL_LIBCPP`: https://github.com/zephyrproject-rtos/zephyr/blob/5f418f506c717ed39686ab0b11a3e6e4bf4fb756/lib/libc/Kconfig#L45-L48

Reply:
That pico one is different, as you can't build from the *source module* today if you need the c++ std lib.

Anyhow, to me it seems we should just set MINIMAL_LIBC_SUPPORTED to n for that board, and others which may just not be able to be built with minimal.
But if others think otherwise I'm not going to oppose this change.